src/Workspaces/Remote/Core/AbstractAssetProvider.cs

```diff
@@ -19,28 +19,29 @@ internal abstract class AbstractAssetProvider
     /// <summary>
     /// return data of type T whose checksum is the given checksum
     /// </summary>
-    public abstract ValueTask<T> GetAssetAsync<T>(Checksum checksum, CancellationToken cancellationToken);
+    public abstract ValueTask<T> GetAssetAsync<T>(ProjectId? hintProject, Checksum checksum, CancellationToken cancellationToken);
```

Comment:
>ProjectId? hintProject,  [](http://example.com/codeflow?start=50&length=24)

If you see other hint data coming down the line, would it make sense for this to be a struct so everywhere piping it through wouldn't need to change?

Reply:
that happens in the followup :)